### DISCUSSION TITLE
Invalid element coordinates when adding to Canvas.

### DISCUSSION THREAD

**Discussion by IVANKIRICHENKO04** [AvaloniaUI/Avalonia#14323] [Q&A]
When I add an element with RotateTransform = 45 property to a Canvas using Canvas.SetLeft/Top, I expect a different position for the element, based on the documentation. Details in the screenshot.
![Uploading Безымянный.png…]()

**Comment by IVANKIRICHENKO04** [comment]
![Безымянный](https://github.com/AvaloniaUI/Avalonia/assets/84570687/bcf3dd67-a471-4047-9f37-62982188bbaa)

**Reply by IVANKIRICHENKO04** [reply]
dots mark cursor positions, adding an element
Canvas.SetLeft(rectangle, lastMousePosition.X);
Canvas.SetTop(rectangle, lastMousePosition.Y);

**Comment by timunie** [comment]
RenderTransform or LayoutTransform?

**Comment by IVANKIRICHENKO04** [comment]
![image](https://github.com/AvaloniaUI/Avalonia/assets/84570687/e335570b-96e6-4c9c-b67c-31823479f124)
RenderTransform

**Reply by timunie** [reply]
RenderTransform is not the right chioce here or you need to caclulate the rotated coordinates on your own. Use `LayoutTransformControl` as a wrapper to get the layout rotated.
